packages/vasp/package.py

```diff
@@ -0,0 +1,309 @@
+            else:
+                make_include = join_path("arch", base_name + "gnu")
+        elif spec.satisfies("%intel"):
+            if spec.satisfies("+openmp"):
+                make_include = join_path("arch", base_name + "intel_omp")
+            else:
+                make_include = join_path("arch", base_name + "intel")
+
+            # A workaround to get fpp
+            filter_file('fpp', self.compiler.fc, make_include)
+            filter_file('ifort -f_com', 'fpp -f_com', make_include)
+        elif spec.satisfies("%nvhpc"):
+            if spec.satisfies("@6.3:"):
+                if spec.satisfies("+openmp"):
+                    make_include = join_path("arch", base_name + "nvhpc_omp_acc")
+                else:
+                    make_include = join_path("arch", base_name + "nvhpc_acc")
+                filter_file("nvfortran", spack_fc, make_include)
+            else:
+                make_include = join_path("arch", base_name + "nv_acc")
+        elif spec.satisfies("%aocc"):
+            if spec.satisfies("@6.3:"):
+                if spec.satisfies("+openmp"):
+                    make_include = join_path("arch", base_name + "aocc_ompi_aocl_omp")
+                else:
+                    make_include = join_path("arch", base_name + "aocc_ompi_aocl")
+            else:
+                if spec.satisfies("+openmp"):
+                    copy(
+                        join_path("arch", base_name + "gnu_omp"),
+                        join_path("arch", base_name + "aocc_omp"),
+                    )
+                    make_include = join_path("arch", base_name + "aocc_omp")
+                else:
+                    copy(
+                        join_path("arch", base_name + "gnu"),
+                        join_path("arch", base_name + "aocc"),
+                    )
+                    make_include = join_path("arch", base_name + "aocc")
+                filter_file("gcc", "{0} {1}".format(spack_cc, "-Mfree"), make_include, string=True)
+                filter_file("g++", spack_cxx, make_include, string=True)
+                filter_file("^CFLAGS_LIB[ ]{0,}=.*$", "CFLAGS_LIB = -O3", make_include)
+                filter_file("^FFLAGS_LIB[ ]{0,}=.*$", "FFLAGS_LIB = -O2", make_include)
+                filter_file("^OFLAG[ ]{0,}=.*$", "OFLAG = -O3", make_include)
+                filter_file(
+                    "^FC[ ]{0,}=.*$", "FC = {0}".format(spec["mpi"].mpifc), make_include, string=True
+                )
+                filter_file(
+                    "^FCL[ ]{0,}=.*$", "FCL = {0}".format(spec["mpi"].mpifc), make_include, string=True
+                )
+        else:
+            if spec.satisfies("+openmp"):
+                make_include = join_path(
+                    "arch", base_name + "{0}_omp".format(spec.compiler.name)
+                )
+            else:
+                make_include = join_path("arch", base_name + spec.compiler.name)
```

Comment:
This seam to be a lot of repetition that could be factorized. Perhaps for the upstream version not necessarily for the internal one